drivers/video/ov7725.c

```diff
@@ -0,0 +1,651 @@
+}
+
+static int ov7725_stream_stop(const struct device *dev)
+{
+	return 0;
+}
+
+static const struct video_format_cap fmts[] = {
+	{
+		.pixelformat = VIDEO_PIX_FMT_RGB565,
+		.width_min = 640,
+		.width_max = 640,
+		.height_min = 480,
+		.height_max = 480,
+		.width_step = 0,
+		.height_step = 0,
+	},
+	{ 0 }
+};
+
+static int ov7725_get_caps(const struct device *dev,
+			   enum video_endpoint_id ep,
+			   struct video_caps *caps)
+{
+	caps->format_caps = fmts;
+	return 0;
+}
+
+static const struct video_driver_api ov7725_driver_api = {
+	.set_format = ov7725_set_fmt,
+	.get_format = ov7725_get_fmt,
+	.get_caps = ov7725_get_caps,
+	.stream_start = ov7725_stream_start,
+	.stream_stop = ov7725_stream_stop,
+};
+
+static int ov7725_init(const struct device *dev)
+{
+	struct ov7725_data *drv_data = dev->data;
+	struct video_format fmt;
+	uint8_t pid, ver;
+	int ret;
+
+
+	if (drv_data->reset_gpio) {
+		ret = gpio_pin_configure(drv_data->reset_gpio,
+					 drv_data->reset_pin,
+					 GPIO_OUTPUT_ACTIVE |
+					 drv_data->reset_flags);
+		if (ret) {
+			return ret;
+		}
+
+		gpio_pin_set(drv_data->reset_gpio,
+			     drv_data->reset_pin,
+			     0);
+		k_sleep(K_MSEC(1));
+		gpio_pin_set(drv_data->reset_gpio,
+			     drv_data->reset_pin,
+			     1);
+		k_sleep(K_MSEC(1));
+	}
+
+	/* Identify the device. */
+	ret = ov7725_read_reg(dev, OV7725_PID, &pid);
+	if (ret) {
+		LOG_ERR("Unable to read PID");
+		return -ENODEV;
+	}
+
+	ret = ov7725_read_reg(dev, OV7725_VER, &ver);
+	if (ret) {
+		LOG_ERR("Unable to read VER");
+		return -ENODEV;
+	}
+
+	if (OV7725_REVISION != (((uint32_t)pid << 8U) | (uint32_t)ver)) {
+		LOG_ERR("OV7725 Get Vision fail\n");
+		return -ENODEV;
+	}
+
+	/* Device identify OK, perform software reset. */
+	ov7725_write_reg(dev, OV7725_COM7, 0x80);
+
+	k_sleep(K_MSEC(2));
+
+	/* set default/init format VGA RGB565 */
+	fmt.pixelformat = VIDEO_PIX_FMT_RGB565;
+	fmt.width = 640;
+	fmt.height = 480;
+	fmt.pitch = 640 * 2;
+	ret = ov7725_set_fmt(dev, VIDEO_EP_OUT, &fmt);
+	if (ret) {
+		LOG_ERR("Unable to configure default format");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/* Unique Instance */
+static struct ov7725_data ov7725_data_0;
+
+static int ov7725_init_0(const struct device *dev)
+{
+	struct ov7725_data *drv_data = dev->data;
+	char *gpio_name = DT_INST_GPIO_LABEL(0, reset_gpios);
+
+	drv_data->reset_pin = DT_INST_GPIO_PIN(0, reset_gpios),
+	drv_data->reset_flags = DT_INST_GPIO_FLAGS(0, reset_gpios),
+
+	drv_data->i2c = device_get_binding(DT_INST_BUS_LABEL(0));
+	if (drv_data->i2c == NULL) {
+		LOG_ERR("Failed to get pointer to %s device!",
+			DT_INST_LABEL(0));
+		return -EINVAL;
+	}
+
+	if (gpio_name) {
+		drv_data->reset_gpio = device_get_binding(gpio_name);
+		if (drv_data->reset_gpio == NULL) {
+			LOG_ERR("Failed to get pointer to %s device!",
+				gpio_name);
+		}
+	}
+
+	drv_data->i2c_addr = DT_INST_REG_ADDR(0);
+
+	return ov7725_init(dev);
+}
+
+DEVICE_AND_API_INIT(ov7725, DT_INST_LABEL(0), &ov7725_init_0,
```

Comment:
That needs to be converted to the new DT device macro: DEVICE_AND_API_INIT -> DEVICE_DT_INST_DEFINE
cf https://github.com/zephyrproject-rtos/zephyr/pull/30779/commits/fc110b67ea66c77c43565a014583ab00012ff409

Reply:
Change is so fast @@. converted done.